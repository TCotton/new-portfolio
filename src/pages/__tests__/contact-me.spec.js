import React from 'react';
import renderer from 'react-test-renderer';
import ContactMe from '../contact-me';

describe('About Me', () => {

	let component, instance;

	beforeEach( () => {
		component = renderer.create(
			<ContactMe />
		);
		instance = component.getInstance();
	});

	it('renders correctly', () => {
		const tree = renderer
			.create(<ContactMe />)
			.toJSON()
		expect(tree).toMatchSnapshot();
	})

	it('should call shouldComponentUpdate', () => {
		const shouldUpdate = instance.shouldComponentUpdate();
		expect(shouldUpdate).toBe(true);
	});
})
