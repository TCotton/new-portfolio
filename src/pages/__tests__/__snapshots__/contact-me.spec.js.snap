// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactMe renders correctly 1`] = `
Array [
  <header>
    <div>
      <h1>
        <a
          href="/"
        >
          The portfolio and blog of web developer Andy Walpole
        </a>
      </h1>
    </div>
  </header>,
  <div>
    <main>
      <div
        className="banner"
      >
        <section>
          <h2
            className="page-top-title"
          >
            Contact me
          </h2>
        </section>
      </div>
      <div
        className="main-content clearfix form content-text"
        itemScope={true}
        itemType="http://schema.org/ContactPage"
      >
        <div
          className="contact-left-large"
        >
          <div
            className="hidden-desktop"
          >
            <h4
              className="content-title"
            >
              Get in touch left
            </h4>
            <p>
              Andy Walpole
              <br />
              Web Developer
            </p>
            <p>
              <a
                className="underline"
                href="mailto:me@andywalpole.me?Subject=Contact%20from%20portfolio"
              >
                me@andywalpole.me
              </a>
            </p>
          </div>
          <h3
            className="content-title"
          >
            Send a message
          </h3>
          <p
            className="contact-form-success none"
          >
            Thank you for taking the time to fill out the form. I will contact you as soon as I can!
          </p>
          <p
            className="contact-form-failure block"
          >
            The form has not been submitted because of errors. Please review the form error messages and click submit again
          </p>
          <form
            name="contactForm"
            noValidate="novalidate"
            onSubmit={[Function]}
          >
            <span
              className="form-division-blocks"
            >
              
              <label
                className="visuallyhidden"
                htmlFor="name"
              >
                Your full name
              </label>
              <input
                className="form-input"
                id="name"
                maxLength="64"
                name="name"
                onChange={[Function]}
                placeholder="Your full name"
                required="required"
                type="text"
                value=""
              />
            </span>
            <span
              className="form-division-blocks"
            >
              
              <label
                className="visuallyhidden"
                htmlFor="email"
              >
                Your email address
              </label>
              <input
                autoCapitalize="off"
                autoCorrect="off"
                className="form-input"
                id="email"
                maxLength="64"
                name="email"
                onChange={[Function]}
                placeholder="Your email address"
                required="required"
                type="email"
                value=""
              />
            </span>
            <span
              className="form-division-blocks"
            >
              
              <label
                className="visuallyhidden"
                htmlFor="message"
              >
                Please write a message
              </label>
              <textarea
                className="form-textarea"
                cols="10"
                id="message"
                maxLength="1000"
                name="message"
                onChange={[Function]}
                placeholder="Please write a message"
                required="required"
                rows="10"
                value=""
              />
            </span>
            <span
              className="hide"
            >
              <label
                className="hide"
                htmlFor="zipcode"
              >
                zipcode
              </label>
              <input
                className="hide"
                id="zipcode"
                name="zipcode"
                onChange={[Function]}
                placeholder="zipcode"
                type="text"
                value=""
              />
            </span>
            <input
              className="button-contact"
              type="submit"
              value="Send message"
            />
          </form>
        </div>
        <div
          className="contact-right-small"
        >
          <h4
            className="content-title"
          >
            Get in touch
          </h4>
          <p>
            Andy Walpole
            <br />
            Web Developer
          </p>
          <p>
            <a
              className="underline"
              href="mailto:me@andywalpole.me?Subject=Contact%20from%20portfolio"
            >
              me@andywalpole.me
            </a>
          </p>
        </div>
      </div>
      <div>
        <p>
          Count: 
          0
        </p>
        <button
          onClick={[Function]}
          type="button"
        >
          Increment
        </button>
      </div>
      Days working as a full-time web developer
      Cups of coffee before I start work in the morning
      Industry RSS feeds I subscribe to using NewsBlur
      Minutes before I check my Twitter feed after I wake up
      Unit test coverage of this site using Karma and Jasmine
    </main>
    <div>
      container
    </div>
    <footer
      className="footer"
    >
      <div
        className="footer-top"
      >
        <a
          href="https://github.com/TCotton/portfolio"
        >
          <pre
            className="code-preview"
          >
            /**
 						* This entire site is coded in AngularJS and React. Inspect the code on GitHub: https://github.com/TCotton/portfolio
 						*/
 						 angular.module('portfolioApp', ['portfolioApp.blogAdminService','portfolioApp.blogAdminController','portfolioApp.blogCommentsDirective','portfolioApp.blogCommentsService','portfolioApp.blogPagesController','portfolioApp.blogPagesDirective','portfolioApp.blogPagesService','portfolioApp.blogPagesFilter','portfolioApp.blogSidebarController','portfolioApp.contractController','portfolioApp.contractService','portfolioApp.footerController','portfolioApp.footerService','portfolioApp.homepageDirective','portfolioApp.miscDirective','portfolioApp.sideProjectsController','portfolioApp.angularReact','portfolioApp.sitemapController','portfolioApp.wordProjectsController','AppConstants','ngResource','ngSanitize','ngRoute','ngAria','HashBangURLs','portfolioAppConfig','jmdobry.angular-cache','underscore','momentLibrary','helperFunctions','requestTimeout','react','detectLocalStorage','appTemplates'
]).config(function($routeProvider, $httpProvider) {

  $httpProvider.defaults.useXDomain = true;
  delete $httpProvider.defaults.headers.common['X-Requested-With'];

  $routeProvider
    .when('/', {
      templateUrl: 'homepage/main.html',
      title: 'The portfolio and blog of web developer Andy Walpole'
    })
    .when('/work-projects', {
      templateUrl: 'work-projects/my_work.html',
      title: 'Noteworthy work projects from the past 18 months',
      controller: 'WorkPageCtrl as WorkProjPageCtrl'
    })
    .when('/work-projects/:workPage', {
      templateUrl: 'work-projects/work_page.html',
      controller: 'WorkPageCtrl as WorkProjPageCtrl'
    })
 						
          </pre>
        </a>
      </div>
      <div
        className="footer-bottom"
      >
        <p
          className="footer-copyright"
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&copy; Andy Walpole / ",
              }
            }
          />
          <a
            href="/sitemap"
          >
            Sitemap
          </a>
        </p>
      </div>
    </footer>
  </div>,
]
`;
